webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ArchiveApi.scala

```diff
@@ -1003,25 +1020,30 @@ class CheckArchiveServiceImpl(
   def checkTechnique(techInfo: TechniquesInfo, techArchive: TechniqueArchive): IOResult[Unit] = {
 
     // check that the imported technique category is the same than the one already existing, if any
-    techInfo.techniquesCategory.collectFirst {
-      case (TechniqueId(name, _), catId) if (name.value == techArchive.technique.id.name.value) => catId
-    } match {
-      case None        => // technique not present, ok
-        ZIO.unit
-      case Some(catId) =>
-        val existing = catId.getPathFromRoot.map(_.value).tail.mkString("/")
-        val archive  = techArchive.category.mkString("/")
-        if (existing == archive) {
+    def checkTechniqueExistSameCat() = {
+      techInfo.techniquesCategory.collectFirst {
+        case (TechniqueId(name, _), catId) if (name.value == techArchive.technique.id.name.value) => catId
+      } match {
+        case None        => // technique not present, ok
           ZIO.unit
-        } else {
-          Inconsistency(
-            s"Technique '${techArchive.technique.id.serialize}' from archive has category '${archive}' but a " +
-            s"technique with that name already exists in category '${existing}': it must be imported in the " +
-            s"same category, please update your archive."
-          ).fail
-        }
+        case Some(catId) =>
+          val existing = catId.getPathFromRoot.map(_.value).tail.mkString("/")
+          val archive  = techArchive.category.mkString("/")
+          if (existing == archive) {
+            ZIO.unit
+          } else {
+            Inconsistency(
+              s"Technique '${techArchive.technique.id.serialize}' from archive has category '${archive}' but a " +
+              s"technique with that name already exists in category '${existing}': it must be imported in the " +
+              s"same category, please update your archive."
+            ).fail
+          }
+      }
     }
+
+    checkTechniqueExistSameCat()
   }
+
```

Comment:
Instead of wrapping the method body with a more descriptive name, shouldn't it be better renaming `checkTechnique` to `checkTechniqueExistsSameCat` then ?
 Or simply comment on the method ? There already is a line comment but inside the method...

Reply:
I change that b/c I though I would be able to just chain check here, so the shape would be: 
```
def checkTechnique(techInfo: TechniquesInfo, techArchive: TechniqueArchive): IOResult[Unit] = { 
  def checkTechniqueExistSameCat() = { ... }
  def otherCheck() = {... }

  checkTechniqueExistSameCat() *> otherCheck()
```

I was not able to do it b/c that chek need to be done before hand, and there's no simple way to factor out all checks at the same place, but I prefer to let it that way, it will be more simple to add other checks latter on (and the cost in perf/compilation/whatever is negligible)